src/main/webapp/script.js

```diff
@@ -41,26 +42,54 @@ function initCommentSlider() {
  * @return {JSON} comments json object of comment objects
  */
 async function getYouTubeComments(url) {
-  url = extractYouTubeUrl(url);
-  const numComments = document.getElementById(SLIDER_NAME).value;
-  const response = await fetch('/YouTubeComments?url=' + url +
-    '&numComments=' + numComments);
-  const comments = await response.json();
-  return comments;
+  try {
+    url = extractYouTubeUrl(url);
+    const numComments = document.getElementById(SLIDER_NAME).value;
+    const response = await fetch('/YouTubeComments?url=' + url +
+      '&numComments=' + numComments);
+    const comments = await response.json();
+    return comments;
+  } catch (err) {
+    throw new Error('Error retrieving comments.');
```

Comment:
Let's remove the try-catch-throw here since it swallows the underlying reason for the failure to retrieve comments.

(I.e. this won't tell us that retrieving the comments failed with a 500 error).